python/lsst/sims/GalSimInterface/galSimInterpreter.py

```diff
@@ -876,16 +892,27 @@ class carrying all of the information for the object whose image
                     # Don't need angles if not doing silicon sensor.
                     surface_ops = [waves, dcr]
 
-                obj.drawImage(method='phot',
-                              offset=offset,
-                              rng=self._rng,
-                              maxN=int(1e6),
-                              image=image,
-                              sensor=sensor,
-                              surface_ops=surface_ops,
-                              add_to_image=True,
-                              poisson_flux=False,
-                              gain=detector.photParams.gain)
+                if use_fft:
+                    obj.drawImage(method='fft',
+                                  offset=offset,
+                                  image=image,
+                                  sensor=sensor,
+                                  surface_ops=surface_ops,
+                                  add_to_image=True,
+                                  gain=detector.photParams.gain)
+                    image.array[image.array < 0] = 0.
+                    image.addNoise(galsim.PoissonNoise())
```

Comment:
Is the logic here that we're drawing the bright objects first and only using FFT-mode rarely (so no overlapping pixels involved)?  Otherwise, I guess you'd in principle need to worry about incorrectly adding Poisson noise twice to the same pixels?

Reply:
Great catch!  Yes, this wasn't correct.  I switched to make a copy of the image for the fft rendering and then add that to the original after the Poisson noise is added.  8a96af1